javascript/builtins/Float16Array.json

```diff
@@ -0,0 +1,205 @@
+{
+  "javascript": {
+    "builtins": {
+      "Float16Array": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Float16Array",
+          "spec_url": "https://tc39.es/proposal-float16array/#sec-float16array",
+          "tags": [
+            "web-features:typed-arrays"
```

Comment:
This is copied from the Float16Array structure, and then replace relevant links to the proposal. I also remove ecmascript 2015 tag, but let this tag since this is still a typed array. 
The question is, do all these parts need a "proposal specific tag?"

Reply:
Thanks Hamish. I'd like to request two changes:

1. Tag everything with "web-features:float16array", remote "typed-arrays" tag. I think we now use proper groups in web-features and don't need this here.
2. Remove constructor_without_parameters and iterable_allowed. These sub features aren't needed for this one. It is only important for the older typed array types which didn't support these behaviors from the start.

Reply:
@Elchi3 Of course - thanks for the direction. Done.